components/tag/index.css

```diff
@@ -242,7 +262,8 @@ governing permissions and limitations under the License.
 
     /* use negative margin to cancel label to edge margin, then add clear button start margin */
     margin-inline-start: calc(var(--mod-tag-label-to-clear-icon, var(--spectrum-tag-label-to-clear-icon))
-                              - (var(--mod-tag-label-spacing-inline-end, var(--spectrum-tag-label-spacing-inline-end))));
+                              - calc(var(--mod-tag-label-spacing-inline-end, var(--spectrum-tag-label-spacing-inline-end))
+                                    - var(--mod-tag-border-width, var(--spectrum-tag-border-width))));
```

Comment:
Since the label to edge margin had 1px removed to account for the border-width, this is subtracted here... this is a little complicated, and operationally turns into `label-to-clear-icon` - `label-spacing-inline-end` + `-border-width`. I think it's maybe slightly easier to reason through with a `calc()` inside the `calc()` but we definitely can do it another way if that seems easier.